### PR TITLE
X-Ray Balance

### PR DESCRIPTION
**Pull request by PKPenguin321**
New gene: Thermal vision
:cl:
balance: X-Ray vision can no longer be obtained via a random gene sequence at roundstart.
balance: X-Ray now has 35 instability, up from 25.
add: A new gene, thermal vision, has been added to genetics.
add: You can create X-Ray vision by mixing Thermal Vision and Radioactive together.
/:cl:

Requiring the gene to be mixed:
- Further timegates it
- Reduces the odds that it will be available in a given round of `n` length
- Means that you will NOT be able to circumvent the instability, as you will never have the gene naturally

Bumping the instability:
- Means X-Ray now takes up a whole 1/3rd of your instability by itself
- Is a serious decision, since you can not circumvent the instability (see above)

Thermal vision is just a strictly worse xray

Untested and currently conflicts since I'm not at my home PC

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
thermal is even better as you can see ppl in smoke

**Comment by PKPenguin321** [comment]
Xray gives the same vision flag and more (xray gives `SEE_TURFS|SEE_MOBS|SEE_OBJS`, my new mutation just gives SEE_MOBS). this is a straight downgrade

**Comment by knacker48** [comment]
How much instability does thermal give? Since its a straight downgrade it should at least cost a little less than X-ray originally costed.

**Comment by necromanceranne** [comment]
Thermal means flash vulnerable though, so I wouldn't call it _better_. Just _pretty good_.

**Comment by 4dplanner** [comment]
How did you decide on 35?

**Comment by PKPenguin321** [comment]
It's 1/3rd of the 100 total safe instability while still being a multiple of 5

**Comment by GuyonBroadway** [comment]
Reminder that timegates basically mean "only powergamers will be bothered to do it". See meth, botany and xenobiology for more information.

**Comment by PKPenguin321** [comment]
You evidently do not know how new genetics works if you believe that to be the case here

**Comment by GuyonBroadway** [comment]
I know exactly how it works and making it a bore to get by hiding it behind a double unlock just means powergamers who want to abuse it are the only ones who shall bother to get it.

And then abuse it in the way those powergamers do which is usually the reason to nerf anything.

**Comment by necromanceranne** [comment]
It's really not that bad honestly. You can get most of the powers with enough time and scans.

File: code/datums/mutations/_combined.dm [hunk 1]
```diff
@@ -17,6 +17,10 @@
 	required = "/datum/mutation/human/strong; /datum/mutation/human/radioactive"
 	result = HULK
 
+/datum/generecipe/x_ray
+	required = "/datum/mutation/human/thermal; /datum/mutation/human/radioactive"
```
Comment (Fox-McCloud) [inline code/datums/mutations/_combined.dm]: using `" "` for typed paths is a sin.

Whoever designed this system to use this should be dragged out into the streets, promptly beaten, stabbed, and finally shot.

**Comment by 4dplanner** [comment]
People get hulk just fine.

**Comment by PKPenguin321** [comment]
Yep. This exact system is already in genetics to gate Hulk, and with enough time the average player can still get it if they want. It in no way makes it "impossible to get unless you powergame," that's just blatantly untrue.

**Review by SpaceManiac — APPROVED** [review]
Seems alright to me. If you have time please take a look at fixing what Fox-McCloud pointed out in a future PR, it really is a maintainability hazard

File: code/datums/mutations/_combined.dm (hunk 1 above)
Comment (Time-Green) [reply code/datums/mutations/_combined.dm]: jfc wow ok
It was originally a list, but they wanted me to do it without initializing it. You can’t use a datums list without Initializing the whole the thing

Comment (Time-Green) [reply code/datums/mutations/_combined.dm]: If you got a better idea, tell me. Simply using a required_1 and required_2 would be a real maintainability hazard

**Comment by PKPenguin321** [comment]
Note to self: update the wiki

**Comment by SpaceManiac** [comment]
Wasn't following the previous PRs, if another maintainer told you otherwise I will try to work it out with them which way is actually better

**Comment by Time-Green** [comment]
> Wasn't following the previous PRs, if another maintainer told you otherwise I will try to work it out with them which way is actually better

@SpaceManiac here's the original [comment](https://github.com/tgstation/tgstation/pull/41258#discussion_r234118809). They just told me to use Initial, the only two ways I could think of doing this was the current string way, or having a variable for each seperate mutation in the mix (wich I didn't use because it would later hinder making mutations that consist of more than two component mutations. But it's not really supported anyway and might not happen)

I'll gladly change it if there's a better way

**Pull request merged**